### PR TITLE
:recycle: (typescript) ported gocardless auth file to TS

### PR DESCRIPTION
**Pull request by MatissJanis**
Just porting `gocardless.js` to typescript + some other files that broke after `gocardless` was ported to TS.

### CODE REVIEW

File: packages/desktop-client/src/gocardless.ts [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
 
     onClose,
     onSuccess,
   });
 }
 
-export async function authorizeBank(pushModal, { upgradingAccountId } = {}) {
+export async function authorizeBank(
+  pushModal: typeof pushModalAction,
+  { upgradingAccountId }: { upgradingAccountId?: string } = {},
```
Comment (joel-jeremy) [inline packages/desktop-client/src/gocardless.ts:40]: Should we extract this to a type?

File: packages/desktop-client/src/gocardless.ts [hunk 2]
```diff
@@ -1,33 +1,44 @@
...
 
-function _authorize(pushModal, upgradingAccountId, { onSuccess, onClose }) {
+function _authorize(
+  pushModal: typeof pushModalAction,
+  upgradingAccountId: string,
+  {
+    onSuccess,
+    onClose,
+  }: {
+    onSuccess: (data: { id: string; accounts: unknown[] }) => Promise<void>;
```
Comment (joel-jeremy) [inline packages/desktop-client/src/gocardless.ts:11]: Should we extract this to a type?

Comment (MatissJanis) [reply packages/desktop-client/src/gocardless.ts]: Makes sense. I've done this now :)

File: packages/desktop-client/src/gocardless.ts (hunk 1 above)
Comment (MatissJanis) [reply packages/desktop-client/src/gocardless.ts:41]: What benefit would that give? It's not really a complex type that we're reusing elsewhere. Just a primitive `string`

Comment (joel-jeremy) [reply packages/desktop-client/src/gocardless.ts:41]: Just to be consistent with the other parts of the codebase but don't consider it as a blocker :)

**Review by joel-jeremy — APPROVED** [review]
Looks good!

**Pull request merged**